plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -662,63 +666,90 @@ def _get_repo_unit_nevra(self, repo_id):
+
     def _get_pkglist_to_publish(self, erratum_unit):
         """
-        Make a list of packages which will be listed in the published erratum.
+        Make a lists of package collections which will be listed in the published erratum.
 
         Only packages contained in the repository being published will be included in this
         errata's pkglist.
 
-        This merges multiple Pulp pkglists into a single pkglist for publication.
+        This merges multiple Pulp pkglists into a single pkglist for publication, unless a
+        module happens. Then a fresh list is used for the module.
 
         :param erratum_unit: the erratum unit to analyze
         :param type: pulp_rpm.plugins.db.models.Errata
-        :return: pkglist to publish
-        :rtype: dict
+        :return: list of pkglist to publish
+        :rtype: list of lists of dicts
         """
+        collection_index = 0
         repo_id = self.get_repo().id
-
-        new_pkglist = {
-            # Due to a quirk in how yum merges pkglists in errata that appear in multiple
-            # repos, the pkglist collection name needs to be unique per-repo when published.
-            # repo_id is already unique and presumably validated, so use that. This is normally a
-            # human-readable string, like "A Human-Readable Repository Description". Pulp doesn't
-            # require storing that sort of string with a repo, so repo_id is the best we've got.
-            'name': repo_id,
-
-            # The repo "short name", like 'repo-name-short'. It also makes sense to use repo_id
-            # here. This appears to be unused by yum when parsing errata, but consistency with the
-            # "long" name field should help in the event that some consumer does use this field.
-            'short': repo_id,
-
-            # Related to the need for a unique collection name, errata should only contain one
-            # pkglist. While filtering packages to include in this errata, add them to this single
-            # list of packages. pkglists can contain multiple collections of packages, but limiting
-            # the published errata to one collection per pkglist and one pkglist per errata per
-            # repository makes the collection name uniqueness requirement easy to fulfill using
-            # the repo_id only.
-            'packages': [],
-
-        }
-
+        # the default, aggregated, non-modular rpms collection
+        new_collection = self._new_collection(repo_id, '{}_default'.format(collection_index))
+        ret = [new_collection]
         if not self.repo_unit_nevra:
             # if repo_unit_nevra is empty, this repo has no RPMs, which makes filtering easy:
             # no pkglist will ever contain packages, so short out and return the empty pkglist
-            return new_pkglist
-
+            # Please note if there's no packages at all, no modules can be associated with a
```

Comment:
I suggest to remove that comment or to rephrase it.
There are valid modules with no rpms in repositories.
In theory it's possible that there are only modules with no rpms and their metadata got updated.
I'm not suggesting to support this case, there is no practical use. But stating that no modules can be associated with a repo because of no rpms doesn't look correct to me. What do you think?

Reply:
> There are valid modules with no rpms in repositories.

ah, didn't know this; I was pretty sure a repo without rpms wouldn't have any modules either, that might need updating by an erratum.  Unsure the logic here is sufficient/sound then.
What would you prefer me to do?